xbmc/music/MusicDatabase.h

```diff
@@ -51,6 +52,33 @@ constexpr int ERROR_REORG_ALBUM = 327;
 constexpr int ERROR_WRITING_CHANGES = 329;
 constexpr int ERROR_COMPRESSING = 332;
 
+/*!
+ * @brief Check if a string contains only digit characters (ignoring leading/trailing whitespace).
+ *
+ * This function trims any leading and trailing whitespace from the input string view,
+ * then checks if all remaining characters are digits (0-9). Returns true if the string
+ * is not empty and all characters are digits, false otherwise.
+ *
+ * @param str The input string view to check.
+ * @return true if the trimmed string is non-empty and contains only digits, false otherwise.
+ */
+constexpr bool IsDigitsOnly(std::string_view str)
```

Comment:
This can be moved to the cpp file in an anonymous namespace.

Reply:
Or entirely be replaced by `StringUtils::IsNaturalNumber`?